pennylane/_device.py

```diff
@@ -285,54 +285,91 @@ def supports_operation(self, operation):
-        if isinstance(observable, type) and issubclass(observable, Observable):
-            return observable.__name__ in self.observables
-        if isinstance(observable, str):
-            return observable in self.observables
+        obs_name = observable
+
+        if issubclass(observable.__class__, Observable):
+            obs_name = observable.name
+        elif isinstance(observable, type):
+            obs_name = observable.__name__
+        elif not isinstance(obs_name, str):
+            raise ValueError("The given observable must either be a pennylane.Observable class or a string.")
+
+        if obs_name not in self.observables:
+            raise DeviceError("Observable {} not supported on device {}".format(obs_name, self.short_name))
 
-        raise ValueError("The given operation must either be a pennylane.Observable class or a string.")
+        return True
+
+    def supports_inverse(self):
+        """Checks if this device supports inverse operations.
+
+        Returns:
+            bool: ``True`` iff inverse operations are supported by the device
+        """
+        if not self.capabilities().get("inverse_operations", False):
+            raise DeviceError("The inverse of gates are not supported on device {}".format(self.short_name))
+        return True
 
     def check_validity(self, queue, observables):
-        """Checks whether the operations and observables in queue are all supported by the device.
+        """Checks whether the operations and observables in queue are all supported by the device. Includes checks for
+        inverse operations.
 
         Args:
             queue (Iterable[~.operation.Operation]): quantum operation objects which are intended
                 to be applied on the device
             observables (Iterable[~.operation.Observable]): observables which are intended
                 to be evaluated on the device
         """
+
         for o in queue:
-            if o.name not in self.operations:
-                raise DeviceError("Gate {} not supported on device {}".format(o.name, self.short_name))
+
+            operation_name = o.name
+
+            if o.inverse:
+                if self.supports_inverse():
+                    operation_name = o.name[:-len(Operation.string_for_inverse)]
+
+            self.supports_operation(operation_name)
 
         for o in observables:
 
             if isinstance(o, Tensor):
-                if "tensor_observables" not in self.capabilities() or not self.capabilities()["tensor_observables"]:
+                if not self.capabilities().get("tensor_observables", False):
                     raise DeviceError("Tensor observables not supported on device {}".format(self.short_name))
 
                 for i in o.obs:
-                    if i.name not in self.observables:
-                        raise DeviceError("Observable {} not supported on device {}".format(i.name, self.short_name))
+                    self.supports_observable(i)
```

Comment:
Is this purely to throw an exception? It's not clear from this line, and not clear from the docstring of `supports_observable`

Reply:
This changed now with the latest version, hopefully it's more apparent now, what is happening there.